superset/jinja_context.py

```diff
@@ -266,6 +338,15 @@ class PrestoTemplateProcessor(BaseTemplateProcessor):
 
     engine = "presto"
 
+    def set_context(self, **kwargs: Any) -> None:
+        super().set_context(**kwargs)
+        self._context[self.engine] = {
```

Comment:
A bit confused here on the parent we have `self._context.update({ ... })` and on Presto we set it inside the engine key, shouldn't they all live on at the dict root level?

Reply:
Apparently these are meant to be prefixed by engine, eg. `presto.latest_partition('my_table')`

Reply:
Yes. The goal was to have the exposed function have the database context (so that they can talk to them without having the user having to specify the `database_id`), but being clear on which method is being used. We have to keep it that way if we care about backwards compatibility.